daemon/cmd/endpoint.go

```diff
@@ -306,25 +306,8 @@ func (m *endpointCreationManager) DebugStatus() (output string) {
 // createEndpoint attempts to create the endpoint corresponding to the change
 // request that was specified.
 func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, epTemplate *models.EndpointChangeRequest) (*endpoint.Endpoint, int, error) {
-	if option.Config.EnableEndpointRoutes {
-		if epTemplate.DatapathConfiguration == nil {
-			epTemplate.DatapathConfiguration = &models.EndpointDatapathConfiguration{}
-		}
-
-		// Indicate to insert a per endpoint route instead of routing
-		// via cilium_host interface
-		epTemplate.DatapathConfiguration.InstallEndpointRoute = true
-
-		// Since routing occurs via endpoint interface directly, BPF
-		// program is needed on that device at egress as BPF program on
-		// cilium_host interface is bypassed
-		epTemplate.DatapathConfiguration.RequireEgressProg = true
-
-		// Delegate routing to the Linux stack rather than tail-calling
-		// between BPF programs.
-		disabled := false
-		epTemplate.DatapathConfiguration.RequireRouting = &disabled
-	}
+	dpConfig := endpoint.NewDatapathConfiguration()
+	epTemplate.DatapathConfiguration = &dpConfig
```

Comment:
The previous code here would override *some* fields in the `models.EndpointDatapathConfiguration` object passed from the cilium-cni, but the new version just creates the datapath configuration from scratch. Was this intentional?

Reply:
Ouch, yes. You are correct and generic-veth uses those other fields.